src/downstream.c

```diff
@@ -69,92 +69,127 @@ void listenerOnMessage(void * msg, size_t msgSize)
+                            resp_msg ->u.req.dest = message->u.req.source;
+                            resp_msg ->u.req.transaction_uuid=message->u.req.transaction_uuid;
+                        }
+                        else
+                        {
+                            resp_msg ->u.crud.source =           message->u.crud.dest;
+                            resp_msg ->u.crud.dest =             message->u.crud.source;
+                            resp_msg ->u.crud.transaction_uuid = message->u.crud.transaction_uuid;
+                            resp_msg ->u.crud.path =             message->u.crud.path;
+                        }
 
-                        ParodusPrint("msgpack encode\n");
-                        resp_size = wrp_struct_to( resp_msg, WRP_BYTES, &resp_bytes );
-                        if(resp_size > 0)
+                        if(response != NULL)
                         {
-                            size = (size_t) resp_size;
-                            sendUpstreamMsgToServer(&resp_bytes, size);
+                            str = cJSON_PrintUnformatted(response);
+                            ParodusInfo("Payload Response: %s\n", str);
+
+                            if( WRP_MSG_TYPE__REQ == msgType )
+                            {
+                                resp_msg ->u.req.payload = (void *)str;
+                                resp_msg ->u.req.payload_size = strlen(str);
+                            }
+                            else
+                            {
+                                resp_msg ->u.crud.payload = (void *)str;
+                                resp_msg ->u.crud.payload_size = strlen(str);
+                            }
+
+                            ParodusPrint("msgpack encode\n");
+                            resp_size = wrp_struct_to( resp_msg, WRP_BYTES, &resp_bytes );
+                            if(resp_size > 0)
+                            {
+                                size = (size_t) resp_size;
+                                sendUpstreamMsgToServer(&resp_bytes, size);
+                            }
+                            free(str);
+                            cJSON_Delete(response);
+                            free(resp_bytes);
+                            resp_bytes = NULL;
                         }
-                        free(str);
-                        cJSON_Delete(response);
-                        free(resp_bytes);
-                        resp_bytes = NULL;
+                        free(resp_msg);
+                        ParodusPrint("free for downstream decoded msg\n");
+                        wrp_free_struct(message);
                     }
-                    free(resp_msg);
-		    ParodusPrint("free for downstream decoded msg\n");
-        	    wrp_free_struct(message);
+                    break;
                 }
+
+                case WRP_MSG_TYPE__EVENT:
+                case WRP_MSG_TYPE__DELETE:
```

Comment:
Why is CRUD DELETE in the default? Should'nt this be with the above CRUD switch case?

Reply:
This is as per design. Only Create/Update and Retrieve of schedule is allowed. Delete not considered, at least for now.

Reply:
This argument may be right from client perspective but we need to think from parodus perspective. Some other client in future might be interested in Delete CRUD message also.